### PR TITLE
add dynamically configurable blacklist param to rack attack

### PR DESCRIPTION
**Pull request by davidsbailey**

### CODE REVIEW

File: lib/cdo/rack/attack.rb [hunk 1]
```diff
@@ -80,6 +80,15 @@ def update_limits
...
     end
+
+    Rack::Attack.set_blacklist_map(blacklist_map)
+  end
+
+  # parse the comma-separated list of channel ids, and return them as a map
+  # which we can use to quickly check whether a given channel id is present.
+  def blacklist_map
+    channels = table_and_property_blacklist.split(',')
+    Hash[channels.map {|channel| [channel, 1]}]
```
Comment (philbogle) [inline lib/cdo/rack/attack.rb]: How about calling this blacklist_set and returning a Set?

Then you can say `Set.new(channels)`here and `@@channel_id_blacklist_set.include?(channel_id)` below.

File: shared/test/test_rack_attack.rb [hunk 2]
```diff
@@ -143,6 +144,8 @@ def test_table_read_limits_with_exponential_backoff_and_logging
     # Now test that a dynamic update to the rate limit is enforced correctly.
     Timecop.freeze rounded_time_now + 60
     _test_table_dynamic_rate_limit_enforced
+    Timecop.freeze rounded_time_now + 60
```
Comment (philbogle) [inline shared/test/test_rack_attack.rb]: The blacklist isn't time dependent is it?

Seems like we could just split out this as a separate test and we don't need Timecop.

**Comment by philbogle** [comment]
LGTM after a couple small comments.

File: shared/test/test_rack_attack.rb (hunk 2 above)
Comment (davidsbailey) [inline shared/test/test_rack_attack.rb]: Good point. This works because Rack Attack checks the blacklist before checking throttles.

**Pull request merged**